airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/version.py

```diff
@@ -0,0 +1,118 @@
+    #     "src/test-performance",
+    #     "build.gradle",
+    #     "erd",
+    #     "build_customization.py",
+    # ]
+
+    def _get_master_metadata(self, connector: Connector) -> Dict[str, Any] | None:
+        """Get the metadata from the master branch or None if unable to retrieve."""
+        github_url_prefix = "https://raw.githubusercontent.com/airbytehq/airbyte/master/airbyte-integrations/connectors"
+        master_metadata_url = f"{github_url_prefix}/{connector.technical_name}/{consts.METADATA_FILE_NAME}"
+        response = requests.get(master_metadata_url)
+
+        # New connectors will not have a metadata file in master
+        if not response.ok:
+            return None
+        return yaml.safe_load(response.text).get("data", None)
+
+    def _parse_version_from_metadata(self, metadata: Dict[str, Any]) -> semver.Version:
+        return semver.Version.parse(str(metadata["dockerImageTag"]))
+
+    def _get_master_connector_version(self, connector: Connector) -> semver.Version:
+        """Get the version from the master branch."""
+        master_metadata = self._get_master_metadata(connector)
+        if not master_metadata:
+            return semver.Version.parse("0.0.0")
+
+        return self._parse_version_from_metadata(master_metadata)
+
+    def _get_current_connector_version(self, connector: Connector) -> semver.Version:
+        """Get the current version."""
+        return self._parse_version_from_metadata(connector.metadata)
+
+    def _are_both_versions_release_candidates(self, master_version: semver.Version, current_version: semver.Version) -> bool:
+        """Check if both versions are release candidates."""
+        return bool(
+            master_version.prerelease
+            and current_version.prerelease
+            and "rc" in master_version.prerelease
+            and "rc" in current_version.prerelease
+        )
+
+    def _have_same_major_minor_patch(self, master_version: semver.Version, current_version: semver.Version) -> bool:
+        """Check if both versions have the same major, minor, and patch versions."""
+        return (
+            master_version.major == current_version.major
+            and master_version.minor == current_version.minor
+            and master_version.patch == current_version.patch
+        )
+
+    def _run(self, connector: Connector) -> CheckResult:
+        """Run the version increment check."""
+        if connector.metadata and connector.metadata.get("ab_internal", {}).get("requireVersionIncrementsInPullRequests") is False:
+            return self.skip(connector, "Connector opts out of version checks.")
+        # if self.context.ci_context == CIContext.MASTER: # TODO: fix this
+        #     return self.skip(
+        #         connector,
+        #         "Connector is in master branch."
+        #     )
+        # if connector.code_directory.modified_files.intersection(self._BYPASS_CHECK_FOR): # TODO: fix this
+        #     return False
```

Comment:
I think we can use a git-based approach to retrieving branch name. Perhaps also to retrieve a files changed list. The poe shell scripts in `poe-tasks` directory have a bash-based example of these actions, I believe. (I had a script to get list of files changed from master, and I think Davin added a script there that uses a check for branch name == master.